massa-execution-worker/src/execution.rs

```diff
@@ -846,12 +887,35 @@ impl ExecutionState {
         )
     }
 
+    /// Gets a data entry both at the latest final and active executed slots
+    ///
+    /// NOTE: temporary, needs to be done in the speculative ledger
+    pub fn get_final_and_active_data_entry(
+        &self,
+        address: &Address,
+        key: &Hash,
+    ) -> (Option<Vec<u8>>, Option<Vec<u8>>) {
+        let final_entry = self.final_state.read().ledger.get_data_entry(address, key);
+        let next_slot = self
+            .active_cursor
+            .get_next_slot(self.config.thread_count)
+            .expect("slot overflow when getting speculative ledger");
+        let search_result = self.fetch_active_history_data_entry(next_slot, address, key);
+        (
+            final_entry.clone(),
+            match search_result {
+                HistorySearchResult::Found(active_entry) => Some(active_entry),
+                HistorySearchResult::NotFound => final_entry,
```

Comment:
In documentation of the API this need to specified that in case of that the value is not in active history it return the same as final and not None.

Reply:
Done.